src/App.tsx

```diff
@@ -535,6 +550,82 @@ class Map {
       }
     }
   }
+
+  getCastlePoints(firstX: number, firstY: number, octaquadrant: number)  {
+    let thisPiece = this.getAt(firstX, firstY).piece
+    console.log(thisPiece.sideConnections[octaquadrant])
+
+    /* if (thisPiece.sideTypes[Math.floor((octaquadrant)/2)] === 2) {
+          return
+        } */
+
+      interface Quadrant {
+        x: number;
+        y: number;
+        quad: number;
+      }
+      let first: Quadrant = {
+        x: firstX,
+        y: firstY,
+        quad: octaquadrant,
+      }
+  
+      let visited: Quadrant[] = [first];
+
+      const pushToVisited = (quad: Quadrant) => {
+        if (!visited.some((extQ) => {
+          return extQ.x === quad.x && extQ.y === quad.y && extQ.quad === quad.quad
+        })) {
+          visited.push(quad)
+        }
+      }
+
+      let index = 0
```

Comment:
```
      let index = 0

      while (visited.length > index) {
      ....
      index ++
```

This would be clearer as for loop

```
for (let index = 0; index < visited.length; index++) {
```